src/containers/AutoUpdateDialog/AutoUpdateDialog.tsx

```diff
@@ -0,0 +1,89 @@
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { checkUpdate, installUpdate, onUpdaterEvent } from '@tauri-apps/api/updater';
+import { UnlistenFn } from '@tauri-apps/api/event';
+import { invoke } from '@tauri-apps/api/tauri';
+import { Button } from '@app/components/elements/Button';
+import Dialog from '@app/components/elements/Dialog';
+import useAppStateStore from '@app/store/appStateStore';
+import { DialogContent } from '../SideBar/components/Settings/Settings.styles';
+import { Typography } from '@app/components/elements/Typography';
+import { ButtonsWrapper } from './AutoUpdateDialog.styles';
+import { CircularProgress } from '@app/components/elements/CircularProgress';
+
+function AutoUpdateDialog() {
+    const setIsAfterAutoUpdate = useAppStateStore((s) => s.setIsAfterAutoUpdate);
+    const [latestVersion, setLatestVersion] = useState<string>();
+    const [isLoading, setIsLoading] = useState(false);
+    const [open, setOpen] = useState(false);
+    const { t } = useTranslation('setup-view', { useSuspense: false });
+
+    useEffect(() => {
+        let unlistenPromise: Promise<UnlistenFn>;
+
+        const checkUpdateTariUniverse = async () => {
+            unlistenPromise = onUpdaterEvent(({ error, status }) => {
+                // This will log all updater events, including status updates and errors.
+                console.info('Updater event', error, status);
+            });
+
+            try {
+                const { shouldUpdate, manifest } = await checkUpdate();
+                if (shouldUpdate) {
+                    console.info('New Tari Universe version available', manifest);
+                    setLatestVersion(manifest?.version);
+                    setOpen(true);
+                } else {
+                    setIsAfterAutoUpdate(true);
+                }
+            } catch (error) {
+                console.error(error);
+            }
+        };
+
+        checkUpdateTariUniverse();
+        return () => {
+            unlistenPromise?.then((unlisten) => unlisten());
+        };
+    }, []);
```

Comment:
I would suggest to always keep the dependency array updated or add ` // eslint-disable-next-line react-hooks/exhaustive-deps` together with a reason why if any variable was not added on purpose.
I'll make a PR trying to make eslint strict for these cases if you agree
```suggestion
    }, [setIsAfterAutoUpdate]);
```